### ISSUE TITLE
Websocket cockpit pending when bitwarden enable

### ISSUE THREAD

**Issue by vuducmanh11** [bitwarden/clients#6817]
### Steps To Reproduce

1. Go to cockpit webserver
2. Click on 'Login' after bitwarden autofill user and password
3. Scroll down cockpit sidebar and click to 'Terminal' tab

### Expected Result

Cockpit terminal display properly

### Actual Result

White screen shell, websocket pending

### Screenshots or Videos

- When enable bitwarden chrome extension
![image](https://github.com/bitwarden/clients/assets/22274112/f0eb7af7-f584-4a62-b6fd-54b895b7869e)
- When disable bitwarden chrome extension
![image](https://github.com/bitwarden/clients/assets/22274112/a012462d-a0e5-4064-a50d-babe26545047)

### Additional Context

I'm using cockpit for accessing ssh from web

### Operating System

Linux

### Operating System Version

CentOS 7

### Web Browser

Chrome

### Browser Version

Version 119.0.6045.105 (Official Build) (64-bit)

### Build Version

2023.10.1

### Issue Tracking Info

- [X] I understand that work is tracked outside of Github. A PR will be linked to this issue should one be opened to address it, but Bitwarden doesn't use fields like "assigned", "milestone", or "project" to track progress.

**Comment by NovaSilentium** [comment]
Hi there,

I am unable to reproduce this issue, it has been escalated for further investigation. If you have more information that can help us, please add it below. 

Thanks!

**Comment by vuducmanh11** [comment]
> Hi there,
> 
> I am unable to reproduce this issue, it has been escalated for further investigation. If you have more information that can help us, please add it below.
> 
> Thanks!

Certainly, I will provide an update if I discover any further information

**Comment by mmcphee** [comment]
Ran into this issue today. When Bitwarden extension is enabled on the browser, and I refresh cockpit, it won't load. When it is disabled, and I refresh the page again, it will load. 
![CockpitError](https://github.com/bitwarden/clients/assets/18193886/2ae23c2c-2f5f-415e-8f86-361f317fbfcc)

**Comment by ratavaz** [comment]
Hi,
I'm having the same issue as [vuducmanh11] with just some minor differences.
In my instance at times after the login I have the same white screen as shown in [vuducmanh11] pictures while other times the screen is completely white, without even the left hand side menu.
When the screen is fully white and I try to reload the page I end up back to the Cockpit login page.

I'm running Cockpit on a Raspberry PI 3
OS is Raspberry PI OS based on Debian 11 (Bullseye)

I can systematically reproduce the problem with:
- Edge Version 119.0.2151.58 (Official build) (64-bit) and previous versions
- Chrome Version 119.0.6045.124 (Official Build) (64-bit) and previous versions
- Firefox Version 119.0.1 (64-bit) and previous versions

Bitwarden Estension version is 2023.10.2

Cockpit is reachable through reverse proxy with ngnix but the same issue is present even if I directly access Cockpit without passing through nginx.

Some additional information:
1. If I use Bitwarden extension to add login and password to the Cockpit login page, then disable the extension before logging in, everything works. If I re-enable the extension at that point everything keeps working, even in case of page reload or logoff and new login. It seems that something messes up Cockpit login procedure but once the login has been completed that specific step is not repeated on subsequent logins.
2. My Cockpit configuration allows me to jump from the server on which Cockpit interface is installed to another server allowing me to control both from the same interface. When I can manage to have Cockpit showing the left hand side menu, if I jump to the second server everything works perfectly with it but the main server keeps giving the white page as to [vuducmanh11].
3. In my instance everything has worked for a while then suddenly stopped, cannot trace back if due to an update to the Bitwarden Extension or to the browser, due to the timing it seems more probable that the trigger was the extension update, but cannot say it for sure.

The only error I can find in the log (journalctl -u cockpit) is the following:
- cockpit-tls[xxxxx]: cockpit-tls: gnutls_handshake failed: A TLS fatal alert has been received.

Any help you could provide is welcome

Best Regards
Paolo

**Comment by Newspring-ple** [comment]
### Explain what happens

**Description:**
I am experiencing an issue with Cockpit when the Bitwarden extension is enabled. After signing in, the page loads to a white screen. This issue occurs consistently across all my Cockpit sessions on different servers. When I disable the Bitwarden extension, Cockpit loads as expected. However, when I enable the Bitwarden extension after Cockpit has loaded, only the left menu bar is available, and the rest of the screen remains white.

**Steps to Reproduce:**
1. Enable the Bitwarden extension in the browser.
2. Log in to Cockpit.
3. Observe that the page loads to a white screen.
4. Disable Bitwarden extension, refresh Cockpit, and observe normal behavior.

**Browser Information:**
- Browser: [Firefox/Chrome]
- Browser Version: 119.0.1 (64-bit)
- Operating System: 
  - Ubuntu "20.04.6 LTS",   
  - Rocky Linux "Rocky Linux 8.8 (Green Obsidian)"
  - Alma Linux  "AlmaLinux 8.5 (Arctic Sphynx)"
- Bitwarden Extension Version: 2023.10.2

**Additional Information:**
- When inspecting the page (F12 -> Console), the following error is observed:
   ```
   Uncaught ReferenceError: cockpit is not defined
   <anonymous> https://10.10.10.12:9090/cockpit/$ac1e990e61826658951196585dd72c7c8bdd7b1c64d37c08342af1919cc258ba/shell/index.js:2
   ```

**Workaround:**
To access Cockpit with the Bitwarden extension enabled, I have been using a private window in Firefox and pasting passwords from the regular Bitwarden extension.

**Additional Notes:**
I have tested this issue on both Firefox and Chrome, and the problem persists.

**Expected Behavior:**
Cockpit should load normally with the Bitwarden extension enabled.

Thank you for your assistance in resolving this issue.


### Version of Cockpit

Version 250.1.1.

**Comment by rivman** [comment]
I have the same issue, Firefox 120.0 (64 bits) and Bit Warden 2023.10.2
White page on Cockpit dashboard after login. Manually  entered credentials or not :

Uncaught ReferenceError: cockpit is not defined 

When disabling the FF Extension, page works. Already put domain in exceptions and all is not blocked in Firefox. 
Thanks.

**Comment by vuducmanh11** [comment]
Today, this bug seems fixed

**Comment by ratavaz** [comment]
> Today, this bug seems fixed

Sadly not for me, I'm still experiencing the bug on Edge, Chrome and Firefox updated to the latest available version via the browsers auto-update and using the latest available Bitwarden extension (Version 2023.12.1), at least according to the browser extension stores.

@vuducmanh11: Could you please let me know which browser and which version of the Bitwarden extension have you tested?

Thanks
Paolo

**Comment by Newspring-ple** [comment]
Sorry, I still am having the same issue also, I am willing to share a zoom
session and show you what it looks like on my end.

[image: image.png]

[image: image.png]

**Comment by Newspring-ple** [comment]
Also does not work on
[image: image.png]
[image: image.png]

**Comment by Newspring-ple** [comment]
[image: image.png]
Has same issue, Sorry for the 3 emails, I just thought I better be
thorough,
Thanks for your help on this.
Let me know if there is any way I can help.

**Comment by closebot-bw** [comment]
⚠️ **Stale Issue Notice**

This issue has been automatically marked as stale due to inactivity. It will be closed in **2 weeks** (October 15, 2025) if no further activity occurs.

If this issue is still relevant and you would like to keep it open, please:
- Comment on this issue to show continued interest
- Provide any additional information or updates
- Confirm that the issue still exists in the latest version

Thank you for your contribution to this project! 🙏

**Comment by closebot-bw** [comment]
🔒 **Issue Closed Due to Inactivity**

This issue has been automatically closed due to lack of activity for an extended period. We periodically review and close inactive issues to help maintain our issue tracker and focus on current priorities.

**If this issue is still relevant:**
- Please create a new issue with updated information
- Include steps to reproduce the problem if it's a bug report
- Mention if this issue still occurs in the latest version

Thank you for your contribution to this project. Your feedback helps us improve! 🙏
